configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -591,6 +591,21 @@ public HttpResponse filedistributionStatus(ApplicationId applicationId, Duration
         return fileReferencesToDelete;
     }
 
+    private void waitForApplicationRemoved(TenantApplications applications, ApplicationId applicationId) {
+        log.log(Level.INFO, "Waiting for " + applicationId + " to be deleted");
+        Duration duration = Duration.ofSeconds(5);
+        Instant end = Instant.now().plus(duration);
+        do {
+            if ( ! (applications.hasApplication(applicationId)))
+                return;
+            log.log(Level.INFO, "Application " + applicationId + " not deleted yet, will retry");
```

Comment:
A bit spammy, but OK for now :)

Reply:
Right, thanks, I'll add a TODO or just remove it tomorrow.